docs/samples/Microsoft.ML.Samples/Dynamic/Trainers/Regression/Gam.cs

```diff
@@ -15,31 +17,35 @@ public static void Example()
             // Read the Housing regression dataset
             var data = DatasetUtils.LoadHousingRegressionDataset(mlContext);
 
-            // Create a pipeline
-            // Concatenate the features to create a Feature vector.
-            // Then append a gam regressor, setting the "MedianHomeValue" column as the label of the dataset,
-            // the "Features" column produced by concatenation as the features column,
-            // and use a small number of bins to make it easy to visualize in the console window.
-            // For real appplications, it is recommended to start with the default number of bins.
             var labelName = "MedianHomeValue";
             var featureNames = data.Schema
                 .Select(column => column.Name) // Get the column names
                 .Where(name => name != labelName) // Drop the Label
                 .ToArray();
-            var pipeline = mlContext.Transforms.Concatenate("Features", featureNames)
-                    .Append(mlContext.Regression.Trainers.Gam(
-                        labelColumnName: labelName, featureColumnName: "Features", maximumBinCountPerFeature: 16));
-            var fitPipeline = pipeline.Fit(data);
 
-            // Extract the model from the pipeline
-            var gamModel = fitPipeline.LastTransformer.Model;
+            // Create a pipeline.
+            var pipeline =
+                // Concatenate the features to create a Feature vector.
+                mlContext.Transforms.Concatenate("Features", featureNames)
+                // Append a GAM regression trainer, setting the "MedianHomeValue" column as the label of the dataset,
+                // the "Features" column produced by concatenation as the features column,
+                // and use a small number of bins to make it easy to visualize in the console window.
+                // For real applications, it is recommended to start with the default number of bins.
+                .Append(mlContext.Regression.Trainers.Gam(labelColumnName: labelName, featureColumnName: "Features", maximumBinCountPerFeature: 16));
 
-            // Now investigate the properties of the Generalized Additive Model: The intercept and shape functions.
+            // Train the pipeline.
+            var trainedPipeline = pipeline.Fit(data);
 
-            // The intercept for the GAM models represent the average prediction for the training data
-            var intercept = gamModel.Bias;
-            // Expected output: Average predicted cost: 22.53
-            Console.WriteLine($"Average predicted cost: {intercept:0.00}");
+            // Extract the model from the pipeline.
+            var gamModel = trainedPipeline.LastTransformer.Model;
+
+            // Now investigate the bias and shape functions of the GAM model.
+
```

Comment:
remove the space below.

Reply:
done

---
In reply to: [266723313](https://github.com/dotnet/machinelearning/pull/2999#discussion_r266723313) [](ancestors = 266723313)